media/com_fabrik/js/tipsBootStrapMock.js

```diff
@@ -194,10 +195,18 @@ define(['jquery', 'fab/fabrik'], function (jQuery, Fabrik) {
 
                     actualWidth = $tip[0].offsetWidth;
                     actualHeight = $tip[0].offsetHeight;
-
-                    switch (inside ? placement.split(' ')[1] : placement) {
+                    var maxwidth = this.options.maxwidth;
+                    var xpos = parseInt(window.fabrikTipXpos);
+                    var tippos = inside ? placement.split(' ')[1] : placement;
+                    switch (tippos) {
                         case 'bottom':
-                            tp = {top: pos.top + pos.height, left: pos.left + pos.width / 2 - actualWidth / 2};
+                            leftpos = pos.left + pos.width / 2 - maxwidth / 2 ;
+                            // If tip is outside viewport, position tip to left/right edge of window (w/margin of 10px) 
+                            if (leftpos < 0) {leftpos = 10;} 
+                            if (leftpos + maxwidth > parseInt(jQuery(window).width())) {
+                                leftpos = parseInt(jQuery(window).width()) - maxwidth -10 ;
```

Comment:
I understand what you are trying to achieve with this PR, but I could really do with an explanation in comments about the algorithm you are using in each case.

Reply:
We are repeating what happened on a previous PR when you argue about review comments rather than just doing what is asked.

If I cannot understand the code i.e. cannot understand the algorithm by reading the code, then I cannot give it a review approval. So if you want your work to be merged, please do what reviewers ask.

Reply:
leftpos is the left position of the popover. If the cursor mouseover position (xpos) is less than (to the left of) leftpos, then move the popover so it is positioned 15px to the left of xpos. The 15px is to account for the arrow which will get positioned at xpos. 

This added feature prevents a narrow popover from being centered above or below a wide element in such a way that the pointer would be centered and no longer be pointing to the label/mouseover position that triggered the tip.